scripts/cpplint.py

```diff
@@ -4614,7 +4614,7 @@ def Check__CPROVER_(filename, clean_lines, linenum, error):
     error: The function to call with any errors found.
   """
   line = clean_lines.lines[linenum]
-  match = Match(r'.*__CPROVER_.*', line)
+  match = Match(r'.*"([^"\\]|\\.)*__CPROVER_([^"\\]|\\.)*".*', line)
```

Comment:
:question: I don't quite understand this regex, probably because I'm not clear on what ends up being escaped in the final regex - it'd be really nice to write a quick test that shows what this does and does not match. Specifically, does it handle:

```
std::string a_really_long_string = "some initial part \
  __CPROVER_gumpf";
```

I suppose it doesn't matter hugely since more likely to do a series of concatenating strings anyway

Reply:
If I am not mistaken, the lint script checks each line separately, so indeed this wouldn't be caught by this regex. Note that this regex also (erroneously) complains about the following:
```
... "foo"; __CPROVER_assume(true); "bar";
``` 
though I don't think this ever appears in practice. Regarding tests, there exists a regression test directory `cpp-linter`. However most of the tests fail on my machine, and it doesn't seem to be included in the standard regression suite.

Reply:
I'd forgotten about those tests, and yup they look quite broken, might try and tidy them up. I was more thinking along the lines of a python unit test - but no worries if not.